platform/linux-generic/arch/aarch64/odp_crypto_armv8.c

```diff
@@ -256,6 +472,50 @@ odp_crypto_session_create(const odp_crypto_session_param_t *param,
 		session->cipher.func = null_crypto_routine;
 		rc = 0;
 		break;
+#if ODP_DEPRECATED_API
+	case ODP_CIPHER_ALG_AES128_GCM:
+		/* AES-GCM requires to do both auth and
+		 * cipher at the same time */
+		if (param->auth_alg != ODP_AUTH_ALG_AES128_GCM)
+			rc = -1;
+#endif
+		/* Fall through */
+	case ODP_CIPHER_ALG_AES_GCM:
+		/* Set cipher mode for AES-GCM */
+		switch (param->cipher_key.length) {
+		case 16:
+			session->cc.mode = AES_GCM_128;
```

Comment:
I think writing cc.mode directly is not good when a setter function is used to set the the field later. This makes the code unnecessarily depend on the cipher constants structure actually having a mode field and makes it appear as if setting the field before calling the setter function would be somehow necessary. The code would be cleaner and easier to understand if there was a local variable mode and only the setter function set cc.mode.

Reply:
Okay, I will make the change. Thanks!